File: pset6/readability/readability.py
from cs50 import get_string

#def main():
#    print(index())
    
def index():
    bookname = get_string("Text: ")
    letter_num = 0
    word_num = 1
    sentence_num = 0
    for i in range(len(bookname)):
        if 'a' <= bookname[i] <= 'z' or 'A' <= bookname[i] <= 'Z':
            letter_num += 1
        elif bookname[i] == ' ':
            word_num += 1
        elif bookname[i] in ['.', '?', '!']:
            sentence_num += 1
    
    L = letter_num / word_num * 100
    S = sentence_num / word_num * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    #round index 
    grade = round(index)
    if (1 <= grade < 16):
        print(f"Grade {grade}")
    elif (grade < 1):
        print("Before Grade 1")
    else:
        print("Grade 16+")
index()           

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string book = get_string("Text: ") ;
    int letter_num = 0;
    int word_num = 1;
    int sentence_num = 0;
    for (int i = 0; i < strlen(book); i++)
    {
        // set letter_num
        if (('a' <= book[i] && book[i] <= 'z') || ('A' <= book[i] && book[i] <= 'Z'))
        {
            letter_num += 1;
        }
        // set book_num
        if (book[i] == ' ')
        {
            word_num += 1;
        }
        //set sentence_num
        if (book[i] == '.' || book[i] == '!' || book[i] == '?')
        {
            sentence_num += 1;
        }
    }
    //calculate index 
    float L = (float)letter_num / (float)word_num * 100;
    float S = (float)sentence_num / (float)word_num * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    // round index 
    int grade = (int) round(index);
    if (1 <= grade && grade < 16)
    {
        printf("Grade %i\n", grade);
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
    //printf("%f letter(s)\n", L);
    //printf("%f word(s)\n", S);
    //printf("%i sentence(s)\n", sentence_num);
    
}